core/modules/system/js/token.js

```diff
@@ -5,77 +5,330 @@
-  attach: function (context, settings) {
-    // Keep track of which textfield was last selected/focused.
-    $(context).find('textarea, input[type="text"]').on('focus', function() {
-      Backdrop.settings.tokenFocusedField = this;
-    });
+    var buffer;
+    var trTemplate;
+    var buttonTemplate;
+    var linkTemplate;
+    var noLinkTemplate;
+    var nameTemplate;
+    var rawTemplate;
+    var descriptionTemplate;
+
+    // Create DOM element templates and assign attributes. 
+    buffer = document.createDocumentFragment();
+    trTemplate = document.createElement('tr');
+    buttonTemplate = document.createElement('button');
+    linkTemplate = document.createElement('a');
+    noLinkTemplate = document.createElement('span');
+    nameTemplate = document.createElement('td');
+    rawTemplate = document.createElement('td');
+    descriptionTemplate = document.createElement('td');
+
+    trTemplate.setAttribute('role', 'row');
+    trTemplate.setAttribute('aria-expanded', 'false');
+    trTemplate.setAttribute('aria-busy', 'false');
+
+    linkTemplate.setAttribute('href', 'javascript:void(0);');
+    linkTemplate.setAttribute('class', 'token-key');
+
+    nameTemplate.setAttribute('role', 'gridcell');
+    nameTemplate.setAttribute('class', 'token-name');
+
+    rawTemplate.setAttribute('role', 'gridcell');
+    rawTemplate.setAttribute('class', 'token-raw');
+
+    descriptionTemplate.setAttribute('role', 'gridcell');
+    descriptionTemplate.setAttribute('class', 'token-description');
 
-    $(context).find('.token-click-insert .token-key').once('token-click-insert', function() {
-      var newThis = $('<a href="javascript:void(0);" title="' + Backdrop.t('Insert this token into your form') + '">' + $(this).html() + '</a>').on('click', function(){
-        if (typeof Backdrop.settings.tokenFocusedField == 'undefined') {
-          alert(Backdrop.t('First click a text field into which the token should be inserted.'));
+    /**
+     * Selects a token.
+     * 
+     * Invoked when a token link is clicked.
+     */
+    function select(event) {
+      var $token = $(event.target);
+
+      event.preventDefault();
+      event.stopPropagation();
+
+      if (Backdrop.settings.tokenBrowserSelectedToken) {
+        Backdrop.settings.tokenBrowserSelectedToken.removeClass('selected-token');
+        Backdrop.settings.tokenBrowserSelectedToken.removeAttr('data-after');
+        Backdrop.settings.tokenBrowserSelectedToken.removeAttr('aria-selected');
+      }
+
+      if (Backdrop.settings.tokenBrowserSelectedToken && $token[0] === Backdrop.settings.tokenBrowserSelectedToken[0]) {
+        Backdrop.settings.tokenBrowserSelectedToken = null;
+      }
+      else {
+        Backdrop.settings.tokenBrowserSelectedToken = $token;
+        Backdrop.settings.tokenBrowserSelectedToken.addClass('selected-token');
+        Backdrop.settings.tokenBrowserSelectedToken.attr('data-after', ' ' + Backdrop.t('(selected)'));
```

Comment:
I think we already have `selected` as a translated string, so let's exclude the parenthesis here.
```suggestion
        Backdrop.settings.tokenBrowserSelectedToken.attr('data-after', ' (' + Backdrop.t('selected') + ')');
```

Reply:
Done.